### PR TITLE
[FX-5154] Migrate TableRow to TailwindCSS

### PR DESCRIPTION
**Pull request by AdrianContiu**
[FX-5154]

### Description

Migration to Tailwind.

### How to test

- [Temploy](https://picasso.toptal.net/fx-5154-migrate-tablerow)
- Check the [production](https://picasso.toptal.net/?path=/story/components-table--table) and compare it with the temploy.

### Screenshots

| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
| Picasso - Table - Google Chrome 2024-04-16 at 5 25 27 PM | Picasso - Table - Google Chrome 2024-04-16 at 9 53 04 PM |
| Picasso - Table - Google Chrome 2024-04-16 at 5 27 44 PM | Picasso - Table - Google Chrome 2024-04-16 at 9 54 39 PM |

### Development checks

- ~Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)~
- ~Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)~
- [x] Make sure that additions and changes on the design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- ~Annotate all `props` in component with documentation~
- ~Create `examples` for component~
- ~Ensure that deployed demo has expected results and good examples~
- ~Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).~
- [x] Self reviewed
- ~Covered with tests ([visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md) included)~

**Breaking change**

- ~codemod is created and showcased in the changeset~
- ~test alpha package of Picasso in StaffPortal~

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-5154]: https://toptal-core.atlassian.net/browse/FX-5154?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by AdrianContiu** [comment]
@toptal-anvil ping reviewers

File: packages/base/Table/src/TableRow/TableRow.tsx [hunk 1]
```diff
@@ -36,30 +30,28 @@ export const TableRow = forwardRef<HTMLTableRowElement, Props>(
...
-        })}
+        className={twMerge(
+          `${
+            isBordered ? 'border-0 border-solid border-b border-gray-200' : ''
+          }`,
+          `${stripeEven ? 'bg-gray-100' : ''}`,
+          `${hover ? 'hover:bg-blue-100' : ''}`,
+          `${selected ? 'bg-blue-100' : ''}`,
+          className
+        )}
```
Comment (TomasSlama) [inline packages/base/Table/src/TableRow/TableRow.tsx:49]:
```suggestion
        className={twMerge(
          isBordered && 'border-0 border-solid border-b border-gray-200',
          stripeEven && 'bg-gray-100',
          hover && 'hover:bg-blue-100' ,
          selected && 'bg-blue-100',
          className
        )}
```

Comment (TomasSlama) [reply packages/base/Table/src/TableRow/TableRow.tsx:49]: According to [the documentation](https://github.com/dcastil/tailwind-merge/blob/v2.2.2/docs/features.md#supports-conditional-classes) it is recommended to use simple condition

**Review by TomasSlama — COMMENTED** [review]
Looks good, tested locally and works also good 💪 I left only one comment that could make the code cleaner in my opinion

**Review by TomasSlama — APPROVED** [review]

File: packages/base/Table/src/TableRow/TableRow.tsx [hunk 2]
```diff
@@ -36,30 +30,26 @@ export const TableRow = forwardRef<HTMLTableRowElement, Props>(
...
         ref={ref}
-        classes={muiClasses}
-        className={cx(className, {
-          [stripeEvenClass]: stripeEven,
-          [borderedClass]: isBordered,
-        })}
+        className={twMerge(
+          isBordered && 'border-0 border-solid border-b border-gray-200',
+          stripeEven && 'bg-gray-100',
+          hover && 'hover:bg-blue-100',
```
Comment (dmaklygin) [inline packages/base/Table/src/TableRow/TableRow.tsx:44]:
```suggestion
          'hover:bg-blue-100',
          hover && 'bg-blue-100'
```

`hover:` should always be assign, but if "hover" prop is passed we must force set "bg-blue-100"

File: packages/base/Table/src/TableRow/styles.ts [hunk 3]
```diff
@@ -1,35 +0,0 @@
...
-    bordered: {
-      borderBottom: `${sizes.borderWidth} solid ${palette.grey.lighter2}`,
-    },
-    stripeEven: {
-      background: palette.grey.lighter,
-    },
-    hover: {
-      transition: transitions.create('background-color', {
-        duration: transitions.duration.shortest,
-      }),
```
Comment (dmaklygin) [inline packages/base/Table/src/TableRow/styles.ts:32]: did we specify styles for the focus transition option?

File: packages/base/Table/src/TableRow/TableRow.tsx (hunk 2 above)
Comment (dmaklygin) [reply packages/base/Table/src/TableRow/TableRow.tsx:44]: The previous statement is wrong, however, as I can see, we anyway should specify "hover:" style for the following states: "selected, hover"

**Review by OleksandrNechai — APPROVED** [review]

File: packages/base/Table/src/TableRow/TableRow.tsx (hunk 2 above)
Comment (AdrianContiu) [reply packages/base/Table/src/TableRow/TableRow.tsx]: I'm not sure that I understand your comment correctly, but I think we don't need to apply the `hover` style for the `selected` state, as in this case, the row will lose its background color when selected and hovered.

**Review by dmaklygin — APPROVED** [review]

**Pull request merged**